diff -uNr lorcon2.orig/lorcon_forge.h lorcon2/lorcon_forge.h
--- lorcon2.orig/lorcon_forge.h	2012-09-06 13:56:55.062128418 -0400
+++ lorcon2/lorcon_forge.h	2012-09-06 13:57:05.298179250 -0400
@@ -35,7 +35,7 @@
 
 
 #ifndef __PACKET_ASSEMBLY_H__
-#include <lorcon_packasm.h>
+#include <lorcon2/lorcon_packasm.h>
 #endif
 
 /* Create a random MAC address, optionally seeded with a valid wireless OUI
diff -uNr lorcon2.orig/Makefile.in lorcon2/Makefile.in
--- lorcon2.orig/Makefile.in	2012-09-06 13:56:55.066128437 -0400
+++ lorcon2/Makefile.in	2012-09-06 13:57:05.298179250 -0400
@@ -66,6 +66,7 @@
 	install -m 644 lorcon_packet.h $(INCLUDE)/lorcon2/lorcon_packet.h
 	install -m 644 lorcon_packasm.h $(INCLUDE)/lorcon2/lorcon_packasm.h
 	install -m 644 lorcon_forge.h $(INCLUDE)/lorcon2/lorcon_forge.h
+	install -m 644 lorcon_endian.h $(INCLUDE)/lorcon2/lorcon_endian.h
 	install -m 644 ieee80211.h $(INCLUDE)/lorcon2/lorcon_ieee80211.h
 	install -d -m 755 $(MAN)/man3
 	install -o root -m 644 lorcon.3 $(MAN)/man3/lorcon.3
diff -uNr lorcon2.orig/ruby-lorcon/Lorcon2.c lorcon2/ruby-lorcon/Lorcon2.c
--- lorcon2.orig/ruby-lorcon/Lorcon2.c	2012-09-06 13:56:55.062128418 -0400
+++ lorcon2/ruby-lorcon/Lorcon2.c	2012-09-06 14:00:50.563296177 -0400
@@ -127,11 +127,11 @@
 
 	if (argc == 2) {
 		rb_scan_args(argc, argv, "2", &rbintf, &rbdriver);
-		intf = STR2CSTR(rbintf);
-		driver = STR2CSTR(rbdriver);
+		intf = StringValuePtr(rbintf);
+		driver = StringValuePtr(rbdriver);
 	} else {
 		rb_scan_args(argc, argv, "1", &rbintf);
-		intf = STR2CSTR(rbintf);
+		intf = StringValuePtr(rbintf);
 	}
 	
 	if (driver == NULL) {
